manager/manager.go

```diff
@@ -66,8 +66,11 @@ type Manager interface {
 	// Gets spec for a container.
 	GetContainerSpec(containerName string) (v2.ContainerSpec, error)
 
-	// Get derived stats for a container.
-	GetContainerDerivedStats(containerName string) (v2.DerivedStats, error)
+	// Get data for all requested containers based on the request options.
+	GetRequestedContainers(containerName string, options v2.RequestOptions) (map[string]*containerData, error)
```

Comment:
We can go back to exposing  manager.GetDerivedStats() and manager.GetSpec() and make containerData private. We'll just need to move that logic from api/ to manager/

Let me do that in this PR.